### ISSUE TITLE
`meson wrap install random` returns a stack trace

### ISSUE THREAD

**Issue by RazrFalcon** [mesonbuild/meson#6769]
**Describe the bug**

`meson wrap install random` returns a stack trace

**To Reproduce**

```
> meson wrap install random
SSL connection failed. Falling back to unencrypted connections.
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "/usr/lib64/python3.6/site-packages/mesonbuild/wrap/wraptool.py", line 216, in run
    options.wrap_func(options)
  File "/usr/lib64/python3.6/site-packages/mesonbuild/wrap/wraptool.py", line 100, in install
    (branch, revision) = get_latest_version(name)
  File "/usr/lib64/python3.6/site-packages/mesonbuild/wrap/wraptool.py", line 83, in get_latest_version
    jd = get_result(API_ROOT + 'query/get_latest/' + name)
  File "/usr/lib64/python3.6/site-packages/mesonbuild/wrap/wraptool.py", line 57, in get_result
    u = open_wrapdburl(urlstring)
  File "/usr/lib64/python3.6/site-packages/mesonbuild/wrap/wrap.py", line 72, in open_wrapdburl
    return urllib.request.urlopen(urlstring, timeout=req_timeout)
  File "/usr/lib64/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/lib64/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.6/urllib/request.py", line 564, in error
    result = self._call_chain(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 756, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib64/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/lib64/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 500: INTERNAL SERVER ERROR
```

**Expected behavior**

A user-friendly error message.

**system parameters**
* Not a cross build
* Gentoo Linux
* Python 3.6.10
* meson 0.52.1

**Comment by eli-schwartz** [comment]
```
$ meson wrap install random

ERROR: WrapDB connection failed to https://wrapdb.mesonbuild.com/v1/query/get_latest/random with error HTTP Error 500: INTERNAL SERVER ERROR
```

Not great, but better. Ultimately seems to be a https://github.com/mesonbuild/mesonwrap bug I think?

**Comment by eli-schwartz** [comment]
Fixed in meson by #6139, should cover 404 HTTPError too (if the wrap server correctly identifies 404).
